Form/FormFlowInterface.php

```diff
@@ -27,9 +27,9 @@ function getName();
 	function setFormFactory(FormFactoryInterface $formFactory);
 
 	/**
-	 * @param Request|null $request
+	 * @param RequestStack $requestStack
 	 */
-	function setRequest(Request $request = null);
+	function setRequestStack(RequestStack $requestStack);
```

Comment:
why ? Setter injection means that your object has to account for setters not being called (and so the property being ``null``). It makes the dependency optional and mutable.